File: Sesion 7/Ejercicio/Ejercicio2 S6/Ejercicio 2.py
# Clase PADRE
class Persona:
    def __init__(self, nombre, edad, salario, email):
        self.nombre = nombre
        self.edad = edad
        self.salario = salario
        self._email = email  # Atributo protegido

# Clases HIJO
class Empleado(Persona):
    def __init__(self, nombre, edad, salario, email):
        super().__init__(nombre, edad, salario, email)

    def get_email(self):
        return self._email

class Cliente(Persona):
    def __init__(self, nombre, edad, salario, email):
        super().__init__(nombre, edad, salario, email)

    def get_email(self):
        return self._email

# Main
emp = Empleado("Mauricio Lazo", 12, 100.50, "maurowo@gmail.com")
cli = Cliente("Sebastián Monteagudo", 50, 20.0, "sebitas123@gmail.com")

print(f"Empleado: {emp.nombre}, Edad: {emp.edad}, Salario: {emp.salario}, Email: {emp.get_email()}")
print(f"Cliente: {cli.nombre}, Edad: {cli.edad}, Salario: {cli.salario}, Email: {cli.get_email()}")

File: Sesion 7/Ejercicio/Ejercicio2 S6/Ejercicio 2.java
// Clase PADRE
class Persona {
    public String nombre;
    public int edad;
    public double salario;
    protected String email;

    public Persona(String nombre, int edad, double salario, String email) {
        this.nombre = nombre;
        this.edad = edad;
        this.salario = salario;
        this.email = email;
    }
}

// Clases HIJO
class Empleado extends Persona {
    public Empleado(String nombre, int edad, double salario, String email) {
        super(nombre, edad, salario, email);
    }

    public String getEmail() {
        return email;
    }
}

class Cliente extends Persona {
    public Cliente(String nombre, int edad, double salario, String email) {
        super(nombre, edad, salario, email);
    }

    public String getEmail() {
        return email;
    }
}

// Main
public class Main {
    public static void main(String[] args) {
        Empleado emp = new Empleado("Mauricio Lazo", 12, 100.50, "maurowo@gmail.com");
        Cliente cli = new Cliente("Sebastián Monteagudo", 50, 20.0, "sebitas123@gmail.com");

        System.out.println("Empleado: " + emp.nombre + ", Edad: " + emp.edad + ", Salario: " + emp.salario + ", Email: " + emp.getEmail());
        System.out.println("Cliente: " + cli.nombre + ", Edad: " + cli.edad + ", Salario: " + cli.salario + ", Email: " + cli.getEmail());
    }
}